src/hotspot/share/gc/parallel/psOldGen.hpp

```diff
@@ -162,6 +165,12 @@ class PSOldGen : public CHeapObj<mtGC> {
   // Iteration.
   void oop_iterate(OopIterateClosure* cl) { object_space()->oop_iterate(cl); }
   void object_iterate(ObjectClosure* cl) { object_space()->object_iterate(cl); }
+  uint iterable_blocks() {
+    return (object_space()->used_in_bytes() + IterateBlockSize - 1) / IterateBlockSize;
+  }
```

Comment:
This causes a build failure on windows:
```
hotspot\share\gc/parallel/psOldGen.hpp(169): error C2220: the following warning is treated as an error
hotspot\share\gc/parallel/psOldGen.hpp(169): warning C4267: 'return': conversion from 'size_t' to 'uint', possible loss of data
```
I suggest either changing the return here to `int` and make an explicit cast or `size_t` and then add the cast when used in `claim_and_get_block()`

Reply:
I made the cast in claim_and_get_block() because I think the itrable_blocks() is not possible to return negative values. and it used in claim_and_get_block() to be a signed int only for helping recoginize survior and eden spaces (the block_index). 
BTW, I didn't rebase this patch to master for quite a while,  I will try to rebase it. 
Thanks!

Reply:
You should not rebase, but **merge** with master, to avoid having to force push.

I agree that adding the cast in `claim_and_get_block()` is better, but you still need to update the return value to be `size_t` here. Otherwise you will still fail with the error seen above.

Reply:
Thanks @kstefanj, I am thinking of the way to not losing review information with rebase. seem merge it better. and I will make the size_t change.